manim/mobject/coordinate_systems.py

```diff
@@ -127,56 +171,55 @@ def input_to_graph_point(self, x, graph):
 class Axes(VGroup, CoordinateSystem):
     def __init__(
         self,
-        x_min=None,
-        x_max=None,
-        y_min=None,
-        y_max=None,
+        x_range=None,
+        y_range=None,
+        x_length=config.frame_width - 2,
+        y_length=config.frame_height - 2,
```

Comment:
In my opinion these lengths are really unnecessary because :
1. they seem very random
2. they can be passed via `x/y_axis_config = {"length": <length>}` or `axis_config = {"length": <length>}`
3. `unit_size` of the `x_axis` and `y_axis` is no longer respected (became redundant) because both `length` and `unit_size` shouldn't be passed into `NumberLine`. Try this code for example:
```py
plane = NumberPlane(
            x_range=[-2, 4],
            y_range=[-2, 3],
            axis_config={"include_numbers": True, "unit_size": 0.5},
        )
# print([axis.unit_size for axis in plane.axes]) --> [2.3703703703703702, 1.6]
self.add(plane)
```
No matter what value you pass for `unit_size` via `axis_config` or `x/y_axis_config`, you'll always get the same result.
For the above reasons, I'd suggest to remove the length parameters

Reply:
1. The lengths are specified so that the axes don't take up the entire screen to make them look more visually appealing, but I'm not sure what you mean by random.
2. I think the length of the axes is a very important property to be able to easily modify. Working with axis configs is generally more tedious as opposed to an explicit parameter. There *is* a little redundancy, but I find the current implementation more intuitive.
3. I hadn't noticed this redundancy, but this makes me think that `unit_size` should be removed, not `x_length` and `y_length`.

Reply:
>I'm not sure what you mean by random

I just meant that the `-2` part felt random, as in it could've been any other number. But we have to use some default length anyway, so that's fine



> 3\. this makes me think that `unit_size` should be removed, not `x_length` and `y_length`

None of those should be compromised in favor of the other imo. Both are extremely useful and removing one of them doesn't look good to me. Maybe instead of `x_length` and `y_length` we can introduce the parameters `x_unit` and `y_unit`? It might not seem to make much difference, but it addresses the following points:

4. The default `NumberPlane` has its `x_axis` stretched out a little
```py
plane = NumberPlane()
print(plane.x_axis.unit_size, plane.y_axis.unit_size) 
>>> 1.0158730158730158 1.0
```
As far as I've seen (and used myself), people often use the default `NumberPlane` as a reference to manim's default coordinate system. With the current implementation, users are forced to do `NumberPlane([-config.frame_x_radius, config.frame_x_radius])` to represent the default coord-system. This also forces the use of `plane.c2p(<coords>)` instead of just `coords`. Although it is a good practice to use `plane.c2p`, it seems a bit too much work while working with default coord-system.

5. Setting `unit_size` becomes difficult
If the user wants to set a specific unit size (extremely useful with `LinearTransformation` demos), its tedious with current implementation
```py
x_min, x_max = -4, 4
x_unit = 2
plane = NumberPlane([x_min, x_max], x_length=(x_max - x_min) * x_unit)  
```

6. Visual appeal (also notice something is wrong with the `NumberLine`'s axis ticks:
```py
# master
# plane = NumberPlane(x_min=-5, x_max=6, y_min=-3, y_max=2)
# line = NumberLine(x_min=-config.frame_x_radius, x_max=config.frame_x_radius)
                
# pr branch
plane = NumberPlane([-5, 6], [-3, 2])
line = NumberLine([-config.frame_x_radius, config.frame_x_radius])

line.shift(-line.n2p(0)).shift(DOWN)
line_origin = Dot(line.n2p(0), color=YELLOW)

self.add(plane.set_opacity(0.5), line, line_origin)
```
on master:
![CoordinateSystemsTest_ManimCE_v0 5 0](https://user-images.githubusercontent.com/73361366/115431646-2ac46080-a223-11eb-9df6-4ff88f6c754a.png)
on pr branch:
![CoordinateSystemsTest_ManimCE_v0 5 0](https://user-images.githubusercontent.com/73361366/115431795-521b2d80-a223-11eb-80bd-9ac8fa2eeec0.png)
It doesn't really look good to me that the plane is always stretched out (and with the origin shifted) and you'll have to do extra work setting the `length` as `(max-min)*unit` to set some `unit_size`

Some points might just look like my personal preferences, I'd like to hear others opinions as well :)

Reply:
I looked into the issue with the misplaced ticks on the number line and it arises due to the the ticks being drawn starting from the left and continuing onwards. This makes it seem like the dot is misplaced based on tick placement but it's still at 0.

![image](https://user-images.githubusercontent.com/70682032/115930265-26808900-a457-11eb-8a69-c4196e68bab1.png)

As for the odd looking default axis. It would be possible to make it so that x_length and y_length are by default ```x_max-x_min``` so that `unit_size` is always one initially. However, it should be rather intuitive for a user that if `x_range =[-5, 5, 1]` that an x_length of 10 would result in a unit_size of one, or `x_length = 5` would make the unit size two.